### PR TITLE
Fix aarch64 guest with EGD rng device got stuck during boot

### PR DESCRIPTION
**Pull request by liuyd96**
aarch64 guest with EGD rng device got stuck during boot.
https://bugzilla.redhat.com/show_bug.cgi?id=1983544

Feed the rng device to fix the error.

### CODE REVIEW

**Comment by liuyd96** [comment]
Before
udp/tcp rng related tests failed on aarch64 due to boot issue.
E.g.
```
[root@hpe-apollo80-01-n00 ~]# avocado run --vt-type libvirt --vt-arch aarch64 --vt-machine-type arm64-mmio libvirt_rng.backend_udp.bind_connect_mode.device_assign
JOB ID     : 87a6df7def7eabcb3f73e7d88b0410ea4acd558a
JOB LOG    : /root/avocado/job-results/job-2021-08-19T02.53-87a6df7/job.log
 (1/1) type_specific.io-github-autotest-libvirt.libvirt_rng.backend_udp.bind_connect_mode.device_assign: ERROR: No ipv4 DHCP lease for MAC 52:54:00:13:69:b1 (526.72 s)
RESULTS    : PASS 0 | ERROR 1 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 527.48 s
```

After
```
[root@hpe-apollo80-01-n00 ~]# avocado run --vt-type libvirt --vt-arch aarch64 --vt-machine-type arm64-mmio libvirt_rng.backend_udp
JOB ID     : d4c38132d144ef91c5bf55258ac8be06a88e2bdd
JOB LOG    : /root/avocado/job-results/job-2021-08-19T02.29-d4c3813/job.log
 (1/6) type_specific.io-github-autotest-libvirt.libvirt_rng.backend_udp.bind_connect_mode.hotplug_unplug.positive.no_options: PASS (69.41 s)
 (2/6) type_specific.io-github-autotest-libvirt.libvirt_rng.backend_udp.bind_connect_mode.hotplug_unplug.positive.persistent: PASS (68.10 s)
 (3/6) type_specific.io-github-autotest-libvirt.libvirt_rng.backend_udp.bind_connect_mode.device_assign: PASS (66.17 s)
 (4/6) type_specific.io-github-autotest-libvirt.libvirt_rng.backend_udp.connect_mode.hotplug_unplug.positive.no_options: PASS (67.83 s)
 (5/6) type_specific.io-github-autotest-libvirt.libvirt_rng.backend_udp.connect_mode.hotplug_unplug.positive.persistent: PASS (73.29 s)
 (6/6) type_specific.io-github-autotest-libvirt.libvirt_rng.backend_udp.connect_mode.device_assign: PASS (73.85 s)
RESULTS    : PASS 6 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 419.58 s

[root@hpe-apollo80-01-n00 ~]# avocado run --vt-type libvirt --vt-arch aarch64 --vt-machine-type arm64-mmio libvirt_rng.backend_tcp
JOB ID     : 1e8da78ecffec318c07b97d51a5fb3fcfe10b341
JOB LOG    : /root/avocado/job-results/job-2021-08-19T02.36-1e8da78/job.log
 (1/7) type_specific.io-github-autotest-libvirt.libvirt_rng.backend_tcp.connect_mode.hotplug_unplug.positive.no_options: PASS (161.49 s)
 (2/7) type_specific.io-github-autotest-libvirt.libvirt_rng.backend_tcp.connect_mode.hotplug_unplug.positive.persistent: PASS (111.70 s)
 (3/7) type_specific.io-github-autotest-libvirt.libvirt_rng.backend_tcp.connect_mode.hotplug_unplug.negative.no_random_source: PASS (15.35 s)
 (4/7) type_specific.io-github-autotest-libvirt.libvirt_rng.backend_tcp.connect_mode.device_assign: PASS (94.46 s)
 (5/7) type_specific.io-github-autotest-libvirt.libvirt_rng.backend_tcp.bind_mode.hotplug_unplug.positive.no_options: PASS (131.99 s)
 (6/7) type_specific.io-github-autotest-libvirt.libvirt_rng.backend_tcp.bind_mode.hotplug_unplug.positive.persistent: PASS (175.98 s)
 (7/7) type_specific.io-github-autotest-libvirt.libvirt_rng.backend_tcp.bind_mode.device_assign: PASS (124.87 s)
RESULTS    : PASS 7 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 816.76 s

```

**Comment by liuyd96** [comment]
@chloerh @yanqzhan Please help review it . Thanks.

File: libvirt/tests/src/libvirt_rng.py [hunk 1]
```diff
@@ -411,6 +413,34 @@ def get_rng_device(guest_arch, rng_model):
         else:
             test.fail("Unknown rng model %s" % rng_model)
 
+    def add_udp_random_server():
+        """
+        Feed the udp rng device some data
```
Comment (chunfuwen) [inline libvirt/tests/src/libvirt_rng.py:418]: Maybe add one comment: this is particularly to aarch64 guest

File: libvirt/tests/src/libvirt_rng.py [hunk 2]
```diff
@@ -411,6 +413,34 @@ def get_rng_device(guest_arch, rng_model):
...
+
+                try:
+                    sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+                    sock.bind((udp_ip, udp_port))
+                    for i in range(5):
+                        data, addr = sock.recvfrom(1024)
+                        logging.debug("SEND DATA")
+                        sock.sendto(b"AB" * 1024, addr)
+                except Exception as e:
+                    test.fail("Failed to add udp server: %s" % str(e))
```
Comment (chunfuwen) [inline libvirt/tests/src/libvirt_rng.py:440]: Failed to add udp server --> Failed to connect udp server?

File: libvirt/tests/src/libvirt_rng.py (hunk 1 above)
Comment (liuyd96) [reply libvirt/tests/src/libvirt_rng.py]: ok, got it. Thanks.

**Review by yanqzhan — APPROVED** [review]
LGTM from feature's aspect.

**Review by chloerh — COMMENTED** [review]
Could you merge these 2 commits to 1? Seems they're dealing with issues coming from the same bug. I'd like to suggest to merge them.

**Comment by liuyd96** [comment]
> Could you merge these 2 commits to 1? Seems they're dealing with issues coming from the same bug. I'd like to suggest to merge them.

Done. Thanks.

**Review by chloerh — APPROVED** [review]
Thanks. Let's merge it.

**Pull request merged**